testsuite/shared/src/main/java/org/jboss/as/test/layers/LayersTest.java

```diff
@@ -321,6 +360,27 @@ private static Set<String> appendResult(String title, Result result, StringBuild
         return deltaModules;
     }
 
+    private static String appendExceptionMsg(String existing, String toAppend) {
+        if (!existing.isEmpty()) {
+            existing += '\n';
+        }
+        return existing + toAppend;
+    }
```

Comment:
What do you think if instead of using Strings this is replaced by using a StreamBuilder()? This would reduce memory consumptions